### PR TITLE
Add optional `speed` parameter to open/closeDrawer 

### PR DESCRIPTION
**Pull request by jeremywho**
## Description

I want to control the speed with which the drawer opens/closes.

Velocity does not seem to provide this functionality. 

Adjusting the Animation.spring `speed` parameter does. 

From the RN Animation docs here: https://reactnative.dev/docs/animated#spring

> speed: Controls speed of the animation. Default 12.
> ...
> velocity: The initial velocity of the object attached to the spring. Default 0 (object is at rest).

## Test plan

I manually tested the change my using different values for speed and comparing the results on device.

### CODE REVIEW

File: DrawerLayout.js [hunk 1]
```diff
@@ -334,6 +340,7 @@ export default class DrawerLayout extends Component<PropType, StateType> {
       bounciness: 0,
       toValue,
       useNativeDriver: this.props.useNativeAnimations,
+      speed: speed != null ? speed : undefined,
```
Comment (jakub-gonet) [inline DrawerLayout.js:343]: I think this can be written as `speed ?? undefined`

**Review by jakub-gonet — CHANGES_REQUESTED** [review]
Could you update the docs as well?

Besides that and my other comment everything looks great, thanks!

**Comment by jeremywho** [comment]
@jakub-gonet I've update the PR. 

I took a stab at the documentation, feel free to request changes to the copy for that.

File: docs/docs/component-drawer-layout.mdx [hunk 2]
```diff
@@ -64,6 +67,28 @@ function. This attribute is present in the standard implementation already and i
...
+### `open`
+
+open can take an optional `options` parameter, enabling further customization of the open animation.
+
+`options` has two optional properties:
+
+`velocity`: number, the initial velocity of the object attached to the spring. Default 0 (object is at rest).
+`speed`: number, controls speed of the animation. Default 12.
+
+### `close`
```
Comment (jakub-gonet) [inline docs/docs/component-drawer-layout.mdx:83]:
```suggestion
### `closeDrawer(options)`
```

File: docs/docs/component-drawer-layout.mdx [hunk 3]
```diff
@@ -64,6 +67,28 @@ function. This attribute is present in the standard implementation already and i
...
+
+close can take an optional `options` parameter, enabling further customization of the close animation.
```
Comment (jakub-gonet) [inline docs/docs/component-drawer-layout.mdx:85]:
```suggestion
`closeDrawer` can take an optional `options` parameter which is an object, enabling further customization of the close animation.
```

**Review by jakub-gonet — CHANGES_REQUESTED** [review]
Left a few inline comments.

Please add some small example to test it on in the description of the PR and please change some existing example in the repo to use `speed` param if possible.

**Comment by jakub-gonet** [comment]
@jeremywho, what's the status of this?

**Comment by jeremywho** [comment]
@jakub-gonet I've updated the branch.

Does the typescript definition need to be updated manually? It would be nice to have the drawer options in there.

**Comment by jakub-gonet** [comment]
Yes, you'll have to extend drawer types exported from RN.

**Comment by jeremywho** [comment]
Typescript def updated.

**Review by jakub-gonet — APPROVED** [review]
LGTM, thanks!

I can merge it after it passes the CI.

**Pull request merged**